### DISCUSSION TITLE
IP Range: 3.0.0.0/8

### DISCUSSION THREAD

**Discussion by thisLambo** [tesla-android/issue-tracker#4] [General]
I'm wanting to use this in my Tesla, but I am reluctant to use it based on the default IP Ranges.

Is it possible to change the range away from anything in 3/8? 

I'd rather not be using AWS IP Space.

I'd suggest changing the default ranges to be in the Private Ranges. 10.0.0.0/8, 172.16.0.0/12, 192.168.0.0/16.

**Comment by thisLambo** [comment]
I just saw your tweet reply to the same question: 
https://twitter.com/mikegapinski/status/1523044677553131520

Can you add this be added to the documentation to clear it up? Thanks!

**Reply by mikegapinski** [reply]
Sure, I'll add that.

Do you have any suggestions for a public IP range that won't make problems?(excluding things listed in https://tools.ietf.org/html/rfc1918?) I used the 3/8 since January and had no problems with working in the car on my MacBook using the Tesla Android hostpot. Prime Video worked, all the Tesla stuff including software updates, sentry etc did too. 

Changing the default ranges is a small change in pihole config and flavor config in the Flutter app, if we figure out a decent alternative I'll do it with the next release

**Reply by thisLambo** [reply]
Actually, does the browser support the CGNAT range (100.64.0.0/10)? It's not RFC1918 at least.

**Reply by mikegapinski** [reply]
Added: https://github.com/tesla-android/tesla-android.github.io/commit/a9a2adae7e564fb31d069f6e37602daf1b75dcae

The CGNAT range should work, however I don't have a clue if using it it won't break anything else. Will check!

I was thinking about using a public range that is not used by 99.999999% of us like something assigned to a small university in the middle of nowhere.

**Reply by HenryLoenwind** [reply]
9.0.0.0/8 is IBM, most of it is used for their intranet. Those should be relatively safe, i.e. not be used for internet services and not open for random people to hijack.

**Reply by elyscape** [reply]
There are [a number of other networks](https://en.wikipedia.org/wiki/Reserved_IP_addresses) allocated but largely unused. Some specifics that are probably worth testing:
- `192.0.2.0/24` - TEST-NET-1
- `198.51.100.0/24` - TEST-NET-2
- `203.0.113.0/24` - TEST-NET-3
- `255.0.0.0/24` - Reserved for future use as part of `240.0.0.0/4`, but unlikely to ever actually be used because of its resemblance to subnet masks

**Reply by elyscape** [reply]
As additional data points, I know of at least [one commercial product][fyde] that uses `255.0.0.0/8`, I've definitely seen `192.0.2.0/24` (TEST-NET-1) used somewhere, and [Tailscale][tailscale] uses `100.64.0.0/10` (CGNAT).

[fyde]: https://www.barracuda.com/products/cloudgen-access
[tailscale]: https://tailscale.com

**Comment by racingsnake91** [comment]
It's really bad practice under any circumstances to use IP space you don't own. I would instead recommend you use one of the private address space ranges for the internal addressing (the addresses given out by the pi acting as the router) and on the Linux/router Pi add a public IP **that you currently manage** bound to the local loopback interface and use this for flutter. This ensures you only actually use a single public IP, and one on which you control what is presented to the wider internet. 

Doing it this way means you could actually run a website on the public IP with a nice DNS record that inside the TeslaAndroid system loads the app, but outside it tells people about the project and how to build it. Furthermore it prevents someone from running a malicious site on the current stolen IP in the future that could potentially be loaded in peoples cars when they're accidentally on LTE not connected to the Pi's wifi.

**Reply by pjktheking** [reply]
全让你懂完了👍
